tests/Console/UpdateSignupAndPostCampaignIdsCommandTest.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+namespace Tests\Console;
+
+use Carbon\Carbon;
+use Tests\TestCase;
+use Rogue\Models\Post;
+use Rogue\Models\Signup;
+use Rogue\Models\Campaign;
+
+class UpdateSignupAndPostCampaignIdsCommandTest extends TestCase
+{
+
+    public function testUpdatingSignupAndPostCampaignIds()
+    {
+        // Create campaign and set a campaign_run_id.
+        $campaign = factory(Campaign::class)->create();
+        $campaign->campaign_run_id = 1;
+        $campaign->save();
+
+        // Create a signup and two posts that belong to this signup with a run_id = 1.
+        $post = factory(Post::class)->create();
+        $post2 = factory(Post::class)->create();
+        $signup = $post->signup;
+
+        $signup->campaign_run_id = 1;
+        $signup->save();
+
+        $post2->signup_id = $signup->id;
+        $post2->save();
+
+        // Run script to update signup and post's campaign id.
+        $this->artisan('rogue:updatesignupandpostcampaignids');
+
+        // Make sure the signup and posts were updated.
+        $this->assertDatabaseHas('signups', [
+            'id' => $signup->id,
+            'northstar_id' => $signup->northstar_id,
+            'campaign_id' => $campaign->id,
+            'campaign_run_id' => 1,
+            'why_participated' => $signup->why_participated,
+        ]);
```

Comment:
Do we know for sure that the `signup` had a `campaign_id` different from `$campaign->id` to start with? I'm thinking if they happened to be the same this could be a false positive.